connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/admin/LoggingResource.java

```diff
@@ -0,0 +1,173 @@
+            Logger current = (Logger) en.nextElement();
+            // exclude loggers that do not have a specific level set
+            if (current.getLevel() != null) {
+                loggers.put(current.getName(), levelToMap(current.getLevel()));
+            }
+        }
+
+        Logger root = LogManager.getRootLogger();
+        if (root.getLevel() != null) {
+            loggers.put(ROOT_LOGGER_NAME, levelToMap(root.getLevel()));
+        }
+
+        return Response.ok(loggers).build();
+    }
+
+    /**
+     * Get the log level of a named logger.
+     *
+     * @param namedLogger name of a logger
+     * @return level of the logger, effective level if the level was not explicitly set.
+     */
+    @GET
+    @Path("/{logger}")
+    public Response getLogger(final @PathParam("logger") String namedLogger) {
+        Objects.requireNonNull(namedLogger, "require non-null name");
+
+        Enumeration en = LogManager.getCurrentLoggers();
+        Logger logger = null;
+        while (en.hasMoreElements()) {
+            Logger l = (Logger) en.nextElement();
+            if (namedLogger.equals(l.getName())) {
+                logger = l;
+                break;
+            }
+        }
+        if (logger == null) {
+            throw new NotFoundException("logger " + namedLogger + " not found.");
+        } else {
+            Map<String, String> levelMap = new HashMap<>();
+
+            String level;
+            if (logger.getLevel() == null) {
+                level = String.valueOf(logger.getEffectiveLevel());
+            } else {
+                level = String.valueOf(logger.getLevel());
+            }
+
+            levelMap.put("level", level);
+            return Response.ok(levelMap).build();
+        }
+    }
+
+
+    /**
+     * Adjust level of a named logger. if name corresponds to an ancestor, then the log level is applied to all child loggers.
+     *
+     * @param namedLogger name of the logger
+     * @param levelMap a map that is expected to contain one key 'level', and a value that is one of the log4j levels:
+     *                 DEBUG, ERROR, FATAL, INFO, TRACE, WARN
+     * @return names of loggers whose levels were modified
+     */
+    @PUT
+    @Path("/{logger}")
+    @SuppressWarnings("unchecked")
+    public Response setLevel(final @PathParam("logger") String namedLogger,
+                             final Map<String, String> levelMap) {
+        String desiredLevelStr = levelMap.get("level");
+        Objects.requireNonNull(desiredLevelStr, "desired level was not specified in PUT request.");
+
+        Level level = Level.toLevel(desiredLevelStr.toUpperCase(Locale.ROOT), null);
+        if (level == null) {
+            throw new NotFoundException("invalid log level '" + desiredLevelStr + "'.");
+        }
+
+        List<Logger> childLoggers;
+        if (ROOT_LOGGER_NAME.equals(namedLogger)) {
+            childLoggers = Collections.list((Enumeration<Logger>) LogManager.getCurrentLoggers());
```

Comment:
Seems like it would be good to have a unit test and is less expensive to test, and in order to do that this class could abstract out the calls to LogManager so that it could be stubbed or mocked in unit tests. It might also be useful to hide the use of Enumeration, especially if that means this class' methods could use more Java 8 streams.